src/downloadlistview.cpp

```diff
@@ -329,52 +336,54 @@ void DownloadListView::keyPressEvent(QKeyEvent* event)
     const int row = qobject_cast<QSortFilterProxyModel*>(model())
                         ->mapToSource(currentIndex())
                         .row();
-    auto state = m_Manager->getState(row);
+    auto& download = m_SourceModel->getDownloadByRow(row);
+    auto state = m_Manager->getState(download.downloadId);
+
     if (state >= DownloadManager::STATE_READY) {
       if (event->key() == Qt::Key_Enter || event->key() == Qt::Key_Return) {
-        issueInstall(row);
+        issueInstall(download.downloadId);
       } else if (event->key() == Qt::Key_Delete) {
-        issueDelete(row);
+        issueDelete(download.downloadId);
       }
     } else if (state == DownloadManager::STATE_DOWNLOADING) {
       if (event->key() == Qt::Key_Delete) {
-        issueCancel(row);
+        issueCancel(download.downloadId);
       } else if (event->key() == Qt::Key_Space) {
-        issuePause(event->key());
+        issuePause(download.downloadId);
       }
     } else if (state == DownloadManager::STATE_PAUSED ||
                state == DownloadManager::STATE_ERROR ||
                state == DownloadManager::STATE_PAUSING) {
       if (event->key() == Qt::Key_Delete) {
-        issueDelete(row);
+        issueDelete(download.downloadId);
       } else if (event->key() == Qt::Key_Space) {
-        issueResume(row);
+        issueResume(download.downloadId);
       }
     }
   }
   QTreeView::keyPressEvent(event);
 }
 
-void DownloadListView::issueInstall(int index)
+void DownloadListView::issueInstall(int downloadId)
```

Comment:
This remarks apply to most function that take a `downloadId` or pass one - Maybe it would be a good idea to pass a `Download const&` instead so that if function need to access the download or one of its field, they can use the download directly.

Reply:
I'm not sure this would be viable in all situations, but it would definitely reduce the calls for the downloadAdded signals. I will look into adding this.